Soccar [Client]/src/nl/soccar/library/SessionController.java

```diff
@@ -63,21 +63,22 @@ public Session join(Session s, String password, Player player) {
             // TODO implementatie foutmelding
             return null;
         }
-        
+
         Team teamBlue = room.getTeamBlue();
         Team teamRed = room.getTeamRed();
+
         List<Player> teamRedPlayers = teamRed.getPlayers();
         List<Player> teamBluePlayers = teamBlue.getPlayers();
-        
-        if (teamBluePlayers.size() < teamRedPlayers.size()) {
-           teamBlue.join(player);
-        } else if (teamBluePlayers.size() > teamRedPlayers.size()) {
-            teamRed.join(player);
-        } else if (rnd.nextInt(2) + 1 == 1) {
+
+        int bluePlayers = teamBluePlayers.size();
+        int redPlayers = teamRedPlayers.size();
+
+        if (bluePlayers < redPlayers || (bluePlayers == redPlayers && (rnd.nextInt(2) + 1 == 1))) {

```

Comment:
Het aanmaken van een random instantie is erg duur, vandaar dat deze als static wordt gemarkeerd.  De random wordt dan ook maar één keer aangemaakt.